share/p2p/shrexnd/client.go

```diff
@@ -0,0 +1,143 @@
+package shrexnd
+
+import (
+	"errors"
+	"fmt"
+
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/net/context"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/ipld"
+	pb "github.com/celestiaorg/celestia-node/share/p2p/shrexnd/pb"
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+	"github.com/celestiaorg/nmt/namespace"
+)
+
+// Client implements client side of shrex/nd protocol to obtain namespaced shares data from remote
+// peers.
+type Client struct {
+	host    host.Host
+	timeout time.Duration
+}
+
+// NewClient creates a new shrEx/nd client
+func NewClient(host host.Host, timeout time.Duration) *Client {
+	return &Client{
+		host:    host,
+		timeout: timeout,
+	}
+}
+
+// GetSharesByNamespace request shares with option to collect proofs from remote peers using shrex
+// protocol
+func (c *Client) GetSharesByNamespace(
+	ctx context.Context,
+	root *share.Root,
+	nID namespace.ID,
+	peerID peer.ID,
```

Comment:
Can we make it variadic? It still allows only one peer to be requested, but after mapping out the implementation of ShrexGetter today, I see the value of having multiple peers requested.

Reply:
this is just the single request, the exported func is variadic, maybe we should rename to `doRequest`?

Reply:
renamed